### PR TITLE
fix:bug that dirty data exists after updating route and wrong mod for prod env

### LINKED ISSUE #690: prod env enable degub mode?
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [+] Bug
- [ ] Requirements
- [ ] Feature or performance improvement
- [ ] Other

___
### Bug
- Which version of Apache APISIX Dashboard, OS, and Browser?
2.0
- What happened?
prod env enable degub mode?
```
func SetUpRouter() *gin.Engine {
	if conf.ENV != conf.EnvLOCAL && conf.ENV != conf.EnvBETA {
		gin.SetMode(gin.DebugMode)
	} else {
		gin.SetMode(gin.ReleaseMode)
	}
```

### LINKED ISSUE #695: Returned with historical dirty data after updating redirect  from disable to custom
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirements
- [ ] Feature or performance improvement
- [ ] Other

___
### Question
- What do you want to know?

___
### Bug
- Which version of Apache APISIX Dashboard, OS, and Browser?

- What happened?
If possible, provide a way to reproduce the error.

Route redirect (redirect) is returned with historical dirty data after updating from disable to custom.
For example， The script field should not return.
image
image

___
### Requirements or improvement
- Please describe your requirements or improvement suggestions.

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- fix issues
fix #695
fix #690

### CODE REVIEW

File: api/internal/handler/route/route.go [hunk 1]
```diff
@@ -304,6 +304,14 @@ func (h *Handler) Update(c droplet.Context) (interface{}, error) {
 				return handler.SpecCodeResponse(err), err
 			}
 		}
+	} else {
+		//remove exists script
+		script, _ := h.scriptStore.Get(input.Route.ID)
+		if script != nil {
+			if err := h.scriptStore.BatchDelete(c.Context(), strings.Split(input.Route.ID, ",")); err != nil {
+				log.Warnf("try to delete script %s again", input.Route.ID)
```
Comment (membphis) [inline api/internal/handler/route/route.go:312]: should we write the error message to log file?

**Comment by Firstsawyou** [comment]
LGTM.

File: api/internal/handler/route/route.go (hunk 1 above)
Comment (nic-chen) [reply api/internal/handler/route/route.go]: sure.but i wrote a wrong log here.

**Comment by juzhiyuan** [comment]
image

It should be `close #xxx, close #xxx`, one number with one keyword.

**Comment by juzhiyuan** [comment]
image

This PR is connected with issues successfully when there has the underline.

**Comment by nic-chen** [comment]
> image
> 
> This PR is connected with issues successfully only when there has an underline.

thanks for tips

**Review by spacewander — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

File: api/internal/handler/route/route.go [hunk 2]
```diff
@@ -309,7 +309,7 @@ func (h *Handler) Update(c droplet.Context) (interface{}, error) {
 		script, _ := h.scriptStore.Get(input.Route.ID)
 		if script != nil {
 			if err := h.scriptStore.BatchDelete(c.Context(), strings.Split(input.Route.ID, ",")); err != nil {
-				log.Warnf("script %s failed", input.Route.ID)
+				log.Warnf("delete script %s failed", input.Route.ID)
```
Comment (membphis) [inline api/internal/handler/route/route.go:312]: I means do we should write the `err` into the error log, the error message is important.

Comment (nic-chen) [reply api/internal/handler/route/route.go:312]: yes, It will write to the error log file.

**Pull request merged**